packages/devtools_app/test/chart_test.dart

```diff
@@ -0,0 +1,931 @@
+          const chartKey = Key('Chart');
+          final controller = ChartController();
+
+          await setupScaledChart(tester, controller, chartKey);
+
+          // Check live view zoom.
+          controller.setXAxisZoomDuration(const Duration());
+          await tester.pumpAndSettle(const Duration(seconds: 2));
+
+          await expectLater(
+            find.byKey(chartKey),
+            matchesGoldenFile('goldens/memory_chart_scaled_live.png'),
+          );
+          // Await delay for golden comparison.
+          await tester.pumpAndSettle(const Duration(seconds: 2));
+
+          // Validate the X axis after data added to all traces.
+          expect(controller.visibleTicks, equals(214));
+          expect(controller.xCanvasChart, equals(50.0));
+          expect(controller.xPaddingRight, equals(0.0));
+          expect(controller.displayXLabels, true);
+          expect(controller.canvasChartWidth, equals(2140.0));
+
+          // Validate the Y axis after data added to all traces.
+          expect(controller.yScale.computedMin, equals(0.0));
+          expect(controller.yScale.computedMax, equals(687573964.4970415));
+          expect(controller.yScale.labelTicks, equals(7.0));
+          expect(controller.yScale.labelUnitExponent, 8.0);
+          expect(controller.yScale.tickSpacing, equals(4142011.834319527));
+          expect(controller.yScale.maxPoint, equals(684801744.0));
+          expect(controller.yScale.maxTicks, equals(170.0));
+
+          final externalTrace = controller.trace(externalTraceIndex);
+          expect(externalTrace.dataYMax, equals(633719952.0));
+          expect(externalTrace.data.length, equals(_rawExternal.length));
+
+          final usedTrace = controller.trace(usedTraceIndex);
+          expect(usedTrace.dataYMax, equals(675253768.0));
+          expect(usedTrace.data.length, equals(_rawUsed.length));
+
+          final capacityTrace = controller.trace(capacityTraceIndex);
+          expect(capacityTrace.dataYMax, equals(684801744.0));
+          expect(capacityTrace.data.length, equals(_rawCapacity.length));
+
+          final rssTrace = controller.trace(rssTraceIndex);
+          expect(rssTrace.dataYMax, equals(571727872.0));
+          expect(rssTrace.data.length, equals(_rawRSS.length));
+
+          expect(controller.timestamps.length, equals(292));
+
+          validateScaledYLabels(controller);
+
+          // Validate the x-axis labels.
+          expect(controller.leftLabelTimestamp, isNotNull);
+          expect(controller.centerLabelTimestamp, isNotNull);
+          expect(controller.rightLabelTimestamp, isNotNull);
+          expect(controller.leftLabelTimestamp, equals(1595682514617));
+          expect(controller.centerLabelTimestamp, equals(1595682534381));
+          expect(controller.rightLabelTimestamp, equals(1595682556248));
+
+          // Only validate if run in a known country, language and timezone.
+          if (isUSLocalePST) {
```

Comment:
remove this if after hard coding the locale and time zone assuming that is easy.

Reply:
won't the golden image tests also have an issue if run in the wrong time zone?

Reply:
I don't think the golden image tests do because the text areas are opaque.